flict/flictlib/format/markdown_format.py

```diff
@@ -10,71 +10,183 @@
-        ret_str += "## Result\n\n"
-        ret_str += "Policy result: "
-        if policy_result == 0:
-            ret_str += "OK"
-        elif policy_result == 1:
-            ret_str += "OK, with licenses to avoid"
-        elif policy_result == 2:
-            ret_str += "Failed identifying outbound license"
-        ret_str += "\n\n"
-        ret_str += "## Suggested outbound licenses \n\n"
-        ret_str += "### Allowed\n\n"
-        for lic in outbounds.get("allowed"):
-            ret_str += " * " + lic + "\n"
-        ret_str += "\n\n"
-        ret_str += "### Avoided\n\n"
-        for lic in outbounds.get("avoided"):
-            ret_str += " * " + lic + "\n"
-        ret_str += "\n\n"
-        ret_str += "### Denied\n\n"
-        for lic in outbounds.get("denied"):
-            ret_str += " * " + lic + "\n"
-
-        return ret_str
+    def format_compatibilities(self, compats):
+        return compats[COMPATIBILITY_TAG]
+
+    def format_licenses(self, licenses):
+        return "\n".join(licenses)
+
+    def parsed_to_license(self, parsed_expr):
+        expr_type = parsed_expr['type']
+
+        if expr_type == 'license':
+            return parsed_expr['name']
+
+        if expr_type == 'operator':
+            name = parsed_expr['name']
+
+            operands = []
+            for op in parsed_expr['operands']:
+                if op['compatibility'] != "Yes":
+                    pass
+                else:
+                    parsed_op = self.parsed_to_license(op)
+                    operands.append(parsed_op)
+            operand_str = f" {name} "
+            expr = operand_str.join(operands)
+            if name == "OR" and len(operands) > 1:
+                expr = f" ( {expr} ) "
+            return expr
+
+    def packages_header(self):
+        return f"{self.headers['packages']} Packages\n"
+
+    def package_name_header(self, name):
+        return f"{self.headers['package_name']} {name}\n"
+
+    def dependencies_header(self):
+        return f"{self.headers['dependencies']} Dependencies\n"
+
+    def problem_header(self):
+        return f"{self.headers['problems']} Problems\n"
+
+    def dependency_header(self, name):
+        return f"{self.headers['dependency']} {name}\n"
+
+    def identified_license_header(self):
+        return f"{self.headers['identified_license']} Identified licenses in the combined work\n"
+
+    def output_compat_markdown(self, compats):
+        # print(str(compats))
+        result = []
+
+        result.append("# License compatibilities\n\n")
+
+        result.append("# Licenses\n\n")
+        for compat in compats['compatibilities']:
+            result.append(" * " + compat['license'] + "\n")
+
+        result.append("\n\n# Compatibilities\n\n")
+        for compat in compats['compatibilities']:
+            main_license = compat['license']
+            result.append(self._output_compat_markdown_licenses(main_license, compat))
+
+        return result
```

Comment:
Why is this returning an array of strings? This could be refactored into a static multiline string.